src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -270,7 +270,7 @@ private boolean isBatchHydrationField(HydrationInputNode hydrationInputNode) {
 
             CompletableFuture<RootExecutionResultNode> serviceResult = serviceExecutor
                     .execute(executionContext, queryTransformationResult, service, operation,
-                            serviceContexts.get(service), true);
+                            serviceContexts.get(service), service.getUnderlyingSchema(), true);
```

Comment:
Note: when `ServiceResultNodesToOverallResult` execution is skipped, now all ERNs point to the overall schema field definition.

This is covering the ARI hook case - the underlying and overall schemas are the same when no transformations are detected, but the `@ARI` directive is only stored on the overall query tree.